low-level/framework/utils/ipmi_client.py

```diff
@@ -252,6 +252,50 @@ def _run_ipmitool_subcommand(self, subcommand, grep_args=None):
 
         return out, error, retcode
 
+    def load_server_fru_list(self):
+        """Get Server FRU list and merge it with server declare fru list
+        maintained in config, with which FRU list can be extended
+        for a solution.
+        
+        Ex: Supermicro servers not listing disk as FRU,
+        though its most common FRU in servers, and 
+        practically it can be replaced easily. 
+        So if for a solution, FRU list needs to be extended
+        beyond what server publishes, 'server_declare_fru' config in sspl.conf
+        can be used. 
+        Some of the usual FRU examples are:- disk, psu, fan."""
+
+        cmd = 'fru list'
+        self.fru_list = []
+        output, err, rc = self._run_ipmitool_subcommand(cmd,
+        grep_args="FRU Device Description : ")
+        if rc != 0:
+            logger.error("Failed in fetching FRU info from server."
+                         f"Error:{err}")
+        if output:
+            for line in output.split('\n'):
+                self.fru_list.append(line.split(': ')[1])
+            keywords = ['Pwr Supply', 'power', 'PS', 'PSU']
+            for match in keywords:
+                for fru in self.fru_list:
+                    if match in fru:
+                        self.fru_list[self.fru_list.index(fru)] = 'psu'
+            self.fru_list = list(dict.fromkeys(self.fru_list))        
```

Comment:
Please use `set` to get unique values out of list. Using `set` is more readable as it gives clear indication that unique values from list is being extracted while `list(dict.fromkeys(self.fru_list))` can be confusing unless one tries it out

Reply:
ok done.